SEDaily-IOS/PodcastDataSource.swift

```diff
@@ -107,24 +107,36 @@ class PodcastDataSource: DataSource {
     }
 
     func update(item: GenericType) {
+        DispatchQueue.global(qos: .userInitiated).async {
+            self.getIndexById(id: item._id, completion: { (index) in
+                guard let index = index else { return }
+                self.getAll(completion: { (podcasts) in
+                    guard var allPodcasts = podcasts else { return }
+                    allPodcasts.remove(at: index)
+                    allPodcasts.insert(item, at: index)
+                    self.override(with: allPodcasts)
+                })
+            })
+        }
+    }
 
+    func override(with items: [GenericType]) {
+        do {
+            try Disk.save(items, to: .caches, as: DiskKeys.PodcastFolder.folderPath)
+        } catch let error {
+            log.error(error.localizedDescription)
+        }
     }
 
-    func clean() {
-        try? Disk.remove(DiskKeys.PodcastFolder.rawValue, from: .caches)
+    static func clean() {
```

Comment:
Why is clean a static function? If you think about it, all of the functions in this class can be static as there's no internal state. Should we consider making all these functions static?

Reply:
Agreed. The datasource can be static. I'll make that change